### ISSUE TITLE
[VsCodeExt] When the backend fails to start, the extension doesn't show the reason - just a timeout

### ISSUE THREAD

**Issue by scale-tone** [scale-tone/DurableFunctionsMonitor#8]
That's because func.exe is being started as a shell.
When started not as a shell, it cannot find env variables needed and thus fails to start.
Need to find a way to either start it directly or to retrieve errors from the shell's stdout.

**Comment by ianadavies** [comment]
I am seeing this on my mac also. Message says to check latest azure functions core tools is installed. I just installed with npm with the following command `npm i -g azure-functions-core-tools` still getting the same error

**Comment by scale-tone** [comment]
Thanks for reporting this, @ianadavies .
Can you go to the extension's install folder (~/.vscode/extensions/durablefunctionsmonitor.durablefunctionsmonitor-2.0.0) and try to manually start the backend with the following commands:

```
func settings decrypt
func settings add AzureWebJobsStorage "your-storage-connection-string"
func start --port 37072
```
?

What happens in that case? Are you then able to navigate to http://localhost:37072/api/monitor ?
Please, also pay attention to your local firewall warnings while doing this.
After doing this experiment, don't forget to remove the created _local.settings.json_ file in that folder.

**Comment by ianadavies** [comment]
Ok, so everything works until I run `func start --port 37072` I then get this output

```backend func start --port 37072
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell]
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell]
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell]

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Azure Functions Core Tools (2.7.1948 Commit hash: 29a0626ded3ae99c4111f66763f27bb9fb564103)
Function Runtime Version: 2.0.12888.0
Can't determine project language from files. Please use one of [--csharp, --javascript, --typescript, --java, --python, --powershell]
[14/01/2020 20:26:12] Building host: startup suppressed: 'False', configuration suppressed: 'False', startup operation id: '4cdfbe02-307b-4443-9805-9326afbb1e7f'
[14/01/2020 20:26:13] Reading host configuration file '/Users/iandavies/.vscode/extensions/durablefunctionsmonitor.durablefunctionsmonitor-2.0.0/backend/host.json'
[14/01/2020 20:26:13] Host configuration file read:
[14/01/2020 20:26:13] {
[14/01/2020 20:26:13]   "version": "2.0"
[14/01/2020 20:26:13] }
[14/01/2020 20:26:13] Reading functions metadata
[14/01/2020 20:26:13] 7 functions found
[14/01/2020 20:26:13] Loading startup extension 'DurableTask'
[14/01/2020 20:26:13] Loaded extension 'DurableTask' (2.0.0.0)
[14/01/2020 20:26:13] A host error has occurred during startup operation '4cdfbe02-307b-4443-9805-9326afbb1e7f'.
[14/01/2020 20:26:13] Microsoft.WindowsAzure.Storage: No valid combination of account information found.
Value cannot be null.
Parameter name: provider
Application is shutting down...
[14/01/2020 20:26:14] Initialization cancellation requested by runtime.
Hosting environment: Production
Content root path: /Users/iandavies/.vscode/extensions/durablefunctionsmonitor.durablefunctionsmonitor-2.0.0/backend
Now listening on: http://0.0.0.0:37072
Application started. Press Ctrl+C to shut down.
[14/01/2020 20:26:14] Stopping host...
[14/01/2020 20:26:14] Host shutdown completed.
```
So I looked at 'local.settings.json' and could see that my connection string was truncated. I wrapped it in `'` single quotes and it then started correctly from the command line.

However it still does not work through the VsCode UI I also tried wrapping the connection string in quotes when prompted by VsCode but still fails with the same error I reported.

I am on mac and don't have my firewall enabled. If I navigate to  http://localhost:37072/api/monitor then UI shows correctly.

**Comment by scale-tone** [comment]
OK, @ianadavies , could you please try a couple of more things.

1. Play a bit with parameters that you can find in **~/.vscode/extensions/durablefunctionsmonitor.durablefunctionsmonitor-2.0.0/out/settings.json** file. Specifically
1.1. Increase **backendTimeoutInSeconds** to e.g. 5 minutes. Does it make any difference?
1.2. Specify the exact port there in **backendBaseUrl**, instead of {portNr} template. Again, any difference?
1.3. Specify 127.0.0.1 instead of localhost in **backendBaseUrl**. Again, any difference?
NOTE: you'll need to restart VsCode after any change you make to that file.

2. While the extension is trying to start the backend, try to navigate to that http://localhost:37072/api/monitor (or whatever you ended up with after step1) URL with your browser or curl. Does it at least respond? Also, can you see the **func.exe** process being started/killed in the meanwhile?

3. If nothing helps, and you're still ready to dedicate your time to figure out the exact problem, then you could get yourself the [extension sources](https://github.com/scale-tone/DurableFunctionsMonitor/tree/master/durablefunctionsmonitor-vscodeext) and debug them - that will show you the actual output and (probably) the error message.

When the extension asks for Connection String, it expects a pure string, no quotes, nothing special, just like this:

> DefaultEndpointsProtocol=https;AccountName=my-account;AccountKey=my-key;EndpointSuffix=core.windows.net

Do you also enter the Hub Name, when it asks? What is it like?
Also, can you please share your VsCode version?

Good news is that you can at least use the tool via browser - the functionality is the same :) Except that the backend endpoint isn't anyhow protected...

**Comment by scale-tone** [comment]
In the meanwhile, the original bug (errors not being shown) has now been fixed in [v2.1](https://marketplace.visualstudio.com/items?itemName=DurableFunctionsMonitor.durablefunctionsmonitor), so closing this.

@ianadavies , if any issues left, please, open a separate bug.
